src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,29 @@ public class ReadWriteList<E> {
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
+        try {
+            list.add(element);
+        } finally {
+            lock.writeLock().unlock();
+        }
     }
 
     public E get(int index) {
-        // write your code here
-        return null;
+        lock.readLock().lock();
+        try {
+            return list.get(index);
+        } finally {
+            lock.readLock().unlock();
+        }
     }
 
     public int size() {
```

Comment:
In the 'size' method, while it is safe to read the size without additional synchronization, it is good practice to document why read locking is used here, as it might not be immediately clear to other developers.